### PR TITLE
plumbing through follow status and getting metadata for full name.

### PR DESCRIPTION
**Pull request by chrisnojima**
also makes search group when adding a participant look correct

### CODE REVIEW

File: shared/actions/chat.js [hunk 1]
```diff
@@ -401,13 +407,61 @@ function * _openFolder (): SagaGenerator<any, any> {
...
+      fullName: metaData.getIn([username, 'fullname'], 'Unknown'),
+    },
+  }))))
   yield put(setSearchActive(true))
 }
 
+function * _updateMetadata (action: UpdateMetadata): SagaGenerator<any, any> {
+  const results: any = yield call(apiserverGetRpcPromise, {
+    param: {
+      endpoint: 'user/lookup',
```
Comment (chrisnojima) [inline shared/actions/chat.js]: when you select a conversation we get metadata on the participants so we can fill in the panel

File: shared/chat/conversation/container.js [hunk 2]
```diff
@@ -23,6 +23,12 @@ class ConversationContainer extends Component<void, Props, State> {
     this.state = {sidePanelOpen: false}
   }
 
+  componentWillReceiveProps (nextProps: Props) {
+    if (this.props.selectedConversation !== nextProps.selectedConversation) {
```
Comment (chrisnojima) [inline shared/chat/conversation/container.js]: lets close the panel if you switch

File: shared/common-adapters/usernames.js [hunk 3]
```diff
@@ -6,11 +6,12 @@ import {isMobile} from '../constants/platform'
 
 import type {Props} from './usernames'
 
-function usernameText ({type, users, style, inline, redColor, backgroundMode}: Props) {
+function usernameText ({type, users, style, inline, redColor, backgroundMode, colorFollowing}: Props) {
```
Comment (chrisnojima) [inline shared/common-adapters/usernames.js]: we color the follow status sometimes

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

**Comment by keybase-ci-visdiff** [comment]
The commits 973da137e216...0d1bd3b70cb0 introduce visual changes on linux. <details><summary>:mag_right: 10 new, 20 changed</summary>

 * Added: **Chat_Side_Panel-Normal**  
   Chat_Side_Panel-Normal rendered
 * Added: **Chat_Side_Panel-Empty**  
   Chat_Side_Panel-Empty rendered
 * Added: **Chat_List-Normal**  
   Chat_List-Normal rendered
 * Added: **Chat_List-Empty**  
   Chat_List-Empty rendered
 * Added: **Chat_Input-Normal**  
   Chat_Input-Normal rendered
 * Added: **Chat_Input-Empty**  
   Chat_Input-Empty rendered
 * Added: **Chat_Header-Normal**  
   Chat_Header-Normal rendered
 * Added: **Chat_Header-Empty** [(view)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Chat_Header-Empty-0d1bd3b70cb0.png)
 * Added: **Chat_Conversations_List-Normal** [(view)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Chat_Conversations_List-Normal-0d1bd3b70cb0.png)
 * Added: **Chat_Conversations_List-Empty** [(view)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Chat_Conversations_List-Empty-0d1bd3b70cb0.png)
 * Changed: **Search-Chat_enabled** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Search-Chat_enabled-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Search-Chat_enabled-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Search-Chat_enabled.png)
 * Changed: **Profile-Your_Profile_Proof_Menu** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Proof_Menu-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Proof_Menu-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Proof_Menu.png)
 * Changed: **Profile-Your_Profile_Pending_Proof_Menu_HN** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Pending_Proof_Menu_HN-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Pending_Proof_Menu_HN-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Pending_Proof_Menu_HN.png)
 * Changed: **Profile-Your_Profile_Pending_Proof_Menu_DNS** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Pending_Proof_Menu_DNS-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Pending_Proof_Menu_DNS-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Pending_Proof_Menu_DNS.png)
 * Changed: **Profile-Your_Profile_Pending_Proof_Menu** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Pending_Proof_Menu-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Pending_Proof_Menu-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Pending_Proof_Menu.png)
 * Changed: **Profile-Your_Profile_No_Proofs** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_No_Proofs-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_No_Proofs-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_No_Proofs.png)
 * Changed: **Profile-Your_Profile_Empty** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Empty-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Empty-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Empty.png)
 * Changed: **Profile-Your_Profile_Broken_Proof_Menu** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Broken_Proof_Menu-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Broken_Proof_Menu-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Broken_Proof_Menu.png)
 * Changed: **Profile-Your_Profile_Broken** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Broken-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Broken-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile_Broken.png)
 * Changed: **Profile-Your_Profile** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Your_Profile.png)
 * Changed: **Profile-Unfollowed_Scrolled** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Unfollowed_Scrolled-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Unfollowed_Scrolled-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Unfollowed_Scrolled.png)
 * Changed: **Profile-Unfollowed_Profile_page** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Unfollowed_Profile_page-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Unfollowed_Profile_page-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Unfollowed_Profile_page.png)
 * Changed: **Profile-Unfollowed_Folders_Expanded** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Unfollowed_Folders_Expanded-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Unfollowed_Folders_Expanded-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Unfollowed_Folders_Expanded.png)
 * Changed: **Profile-Unfollowed_Few_Folders** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Unfollowed_Few_Folders-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Unfollowed_Few_Folders-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Unfollowed_Few_Folders.png)
 * Changed: **Profile-Unfollowed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Unfollowed-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Unfollowed-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Unfollowed.png)
 * Changed: **Profile-Followed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Followed-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Followed-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Followed.png)
 * Changed: **Profile-Changed_Scrolled** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Changed_Scrolled-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Changed_Scrolled-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Changed_Scrolled.png)
 * Changed: **Profile-Changed** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Changed-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Changed-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Profile-Changed.png)
 * Changed: **Menubar-Private_Async_Writing** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Menubar-Private_Async_Writing-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Menubar-Private_Async_Writing-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Menubar-Private_Async_Writing.png)
 * Changed: **Folders_Clear_History_Confirmation-Long_Name_Public** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Folders_Clear_History_Confirmation-Long_Name_Public-973da137e216.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Folders_Clear_History_Confirmation-Long_Name_Public-0d1bd3b70cb0.png) [(diff)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479508988022-973da137e216-0d1bd3b70cb0-linux/Folders_Clear_History_Confirmation-Long_Name_Public.png)
</summary>

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by keybase-ci-visdiff** [comment]
The commits 4022693e454e...b42c23f97a95 introduce visual changes on linux. <details><summary>:mag_right: 2 new, 5 changed</summary>

 * Added: **Chat_Side_Panel-Normal**  
   Chat_Side_Panel-Normal rendered
 * Added: **Chat_Side_Panel-Empty**  
   Chat_Side_Panel-Empty rendered
 * Changed: **Chat_Input-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479753192152-4022693e454e-b42c23f97a95-linux/Chat_Input-Normal-4022693e454e.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479753192152-4022693e454e-b42c23f97a95-linux/Chat_Input-Normal-b42c23f97a95.png)  
   Chat_Input-Normal comparison
 * Changed: **Chat_Input-Empty** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479753192152-4022693e454e-b42c23f97a95-linux/Chat_Input-Empty-4022693e454e.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479753192152-4022693e454e-b42c23f97a95-linux/Chat_Input-Empty-b42c23f97a95.png)  
   Chat_Input-Empty comparison
 * Changed: **Chat_Header-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479753192152-4022693e454e-b42c23f97a95-linux/Chat_Header-Normal-4022693e454e.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479753192152-4022693e454e-b42c23f97a95-linux/Chat_Header-Normal-b42c23f97a95.png)  
   Chat_Header-Normal comparison
 * Changed: **Chat_Header-Empty** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479753192152-4022693e454e-b42c23f97a95-linux/Chat_Header-Empty-4022693e454e.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479753192152-4022693e454e-b42c23f97a95-linux/Chat_Header-Empty-b42c23f97a95.png)  
   Chat_Header-Empty comparison
 * Changed: **Chat_Conversations_List-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479753192152-4022693e454e-b42c23f97a95-linux/Chat_Conversations_List-Normal-4022693e454e.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479753192152-4022693e454e-b42c23f97a95-linux/Chat_Conversations_List-Normal-b42c23f97a95.png)  
   Chat_Conversations_List-Normal comparison
</summary>

File: shared/chat/conversation/header.desktop.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ import type {Props} from './'
 
 const Header = ({participants, onOpenFolder, toggleSidePanel, sidePanelOpen}: Props) => (
   <Box style={containerStyle}>
-    <Usernames inline={false} type='BodyBig' users={participants.filter(p => !p.you).toArray()}
+    <Usernames colorFollowing={true} inline={false} type='BodyBig' users={participants.filter(p => !p.you).toArray()}
```
Comment (gabriel) [inline shared/chat/conversation/header.desktop.js]: This could be empty if only chatting with self

File: shared/actions/chat.js [hunk 5]
```diff
@@ -401,13 +407,61 @@ function * _openFolder (): SagaGenerator<any, any> {
...
+  const metaData = ((yield select(metaDataSelector)): any)
+
+  const following = (yield select(followingSelector)) || {}
+
+  yield put(searchAddUsersToGroup(action.payload.existingParticipants.map(username => ({
+    service: 'keybase',
+    username,
+    isFollowing: !!following[username],
+    extraInfo: {
+      service: 'none',
```
Comment (chrisnojima) [inline shared/actions/chat.js]: We're essentially making a fake search result and only the usernames are really used. I think later we could potentially make a new api endpoint to get the exact search results for all participants but seems like not an MVP thing at the moment.

**Comment by keybase-ci-visdiff** [comment]
The commits 4022693e454e...fb1d69924e9e introduce visual changes on linux. <details><summary>:mag_right: 2 new, 5 changed</summary>

 * Added: **Chat_Side_Panel-Normal**  
   Chat_Side_Panel-Normal rendered
 * Added: **Chat_Side_Panel-Empty**  
   Chat_Side_Panel-Empty rendered
 * Changed: **Chat_Input-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479825663073-4022693e454e-fb1d69924e9e-linux/Chat_Input-Normal-4022693e454e.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479825663073-4022693e454e-fb1d69924e9e-linux/Chat_Input-Normal-fb1d69924e9e.png)  
   Chat_Input-Normal comparison
 * Changed: **Chat_Input-Empty** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479825663073-4022693e454e-fb1d69924e9e-linux/Chat_Input-Empty-4022693e454e.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479825663073-4022693e454e-fb1d69924e9e-linux/Chat_Input-Empty-fb1d69924e9e.png)  
   Chat_Input-Empty comparison
 * Changed: **Chat_Header-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479825663073-4022693e454e-fb1d69924e9e-linux/Chat_Header-Normal-4022693e454e.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479825663073-4022693e454e-fb1d69924e9e-linux/Chat_Header-Normal-fb1d69924e9e.png)  
   Chat_Header-Normal comparison
 * Changed: **Chat_Header-Empty** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479825663073-4022693e454e-fb1d69924e9e-linux/Chat_Header-Empty-4022693e454e.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479825663073-4022693e454e-fb1d69924e9e-linux/Chat_Header-Empty-fb1d69924e9e.png)  
   Chat_Header-Empty comparison
 * Changed: **Chat_Conversations_List-Normal** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479825663073-4022693e454e-fb1d69924e9e-linux/Chat_Conversations_List-Normal-4022693e454e.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1479825663073-4022693e454e-fb1d69924e9e-linux/Chat_Conversations_List-Normal-fb1d69924e9e.png)  
   Chat_Conversations_List-Normal comparison
</summary>

**Pull request merged**